src/algorithm/core.py

```diff
@@ -983,17 +973,20 @@ def draw_field_with_question_meta(
                 evaluation_config,
             )
 
-    def get_occurences(self, question_meta, field_value):
+    def get_matched_answer_groups(self, question_meta, field_value):
         occurences = []
-        if question_meta["answer_type"] == AnswerType.MULTIPLE_CORRECT:
-            for i in range(0, len(question_meta["answer_item"])):
-                if field_value in question_meta["answer_item"][i]:
+        answer_type, answer_item = map(
+            question_meta.get, ["answer_type", "answer_item"]
+        )
+        if answer_type == AnswerType.MULTIPLE_CORRECT:
+            for i in range(0, len(answer_item)):
+                if field_value in answer_item[i]:
                     occurences.append(i)
         else:
-            for i in range(0, len(question_meta["answer_item"])):
+            for i in range(0, len(answer_item)):
```

Comment:
nit: Avoid using range() loops unnecessarily. Loop like this instead -
```suggestion
            for allowed_answer in answer_item:
```